cmake/CMakeLists.txt

```diff
@@ -108,8 +108,11 @@ option(onnxruntime_USE_TELEMETRY "Build with Telemetry" OFF)
 #It's default OFF because it's experimental now.
 option(onnxruntime_PREFER_SYSTEM_LIB "Experimental: Build with the preinstalled libraries in your system" OFF)
 
+# ORT mininal build settings
 option(onnxruntime_MINIMAL_BUILD "Exclude as much as possible from the build. Support ORT format models. No support for ONNX format models." OFF)
 option(onnxruntime_REDUCED_OPS_BUILD "Reduced set of kernels are registered in build via modification of the kernel registration source files." OFF)
+# For now onnxruntime_NO_EXCEPTIONS will only work with onnxruntime_MINIMAL_BUILD, more changes (ONNX, non-CPU EP, ...) are required to run this standalone
+option(onnxruntime_NO_EXCEPTIONS "Disable exception handling." OFF)
```

Comment:
Do we print an error and exit if the user specifies this option without the minimal build flag?

Reply:
So far this options only works with Minimal Build, (without Minimal Build enabled, ORT will not compile with this option on), there are more work needed for this to run standalone, for example, ONNX need to be updated with a no-exception-handling build option, and currently only CPU EP is updated with no-exception-hnadling (all other EPs will still handle exceptions) ...

Reply:
Yeah, we don't have to make this work standalone as I understand there's more work required. Just wondering if we can avoid people from running into build issues and filing support tickets by throwing an error much earlier if they use this option without using the minimal build flag. If these are exposed in build.py, then it's easier to catch such things.

Reply:
Add a fatal message if onnxruntime_NO_EXCEPTIONS is enabled without onnxruntime_NO_EXCEPTIONS

---
In reply to: [480359120](https://github.com/microsoft/onnxruntime/pull/4975#discussion_r480359120) [](ancestors = 480359120)